### ISSUE TITLE
[BUG] AddToCart component doesn't emit click properly

### ISSUE THREAD

**Issue by undercoverGod** [vuestorefront/storefront-ui#1994]
**Describe the bug**
In the component SfAddToCart (molecule), event "click" is used to suggest user has clicked "Add To Cart" button.
This is though convenient logically wrong. If click event listener is attached to this component, default developer's expectation would be to capture click in whole component. But currently it ignores click in areas except "Add To Cart" button.
The add to cart should be captured via a custom event.

**How to reproduce**
1. Create a AddToCart Component
2. Attach a v-on:click method on the component that alerts/logs some message so we know method was called.
3. Click Area between SfButton and SfQuantitySelector

**Expected behavior**
A Click handler method should be called.

**Actual behavior**
Click handler method is not called

**Comment by AdamPawlinski** [comment]
@undercoverGod  Hi, I know what you mean, but I wonder if this change is necessary in such component. I think that rather  most users don't expect that whole component can be clickable (except + and - buttons which have separate event), but only the 'Add to card' button. There is another important thing, it would be a breaking change as it force users to change event name in every component that is already implemented. Thank you for noticing it and let me know what do you think?

**Comment by undercoverGod** [comment]
@AdamPawlinski  I agree with what you have said. I discovered this while working on  #966. As you must be aware this kind of event binding exist in other components as well. Now I think what has been done is maybe due to constraint of working with functional components.  Since template-functional components are no longer supported in Vue 3, we do not have any such constraint in Vue 3.
In this case your second point would become invalid as this package would be consumed by new projects or atleast one who are going to have vue3 migration in which this would be not-so-significant change.
But I am not sure if such behavior difference between current and next version should be done.
Please let me know if you think we should replicate exact behaviour or maybe allow changes of this kind?

**Comment by AdamPawlinski** [comment]
I think it might be a good idea to change this behavior for the Vue3 version, so we can identify other components where this should be done to include in a StorefrontUI that supports Vue3.

**Comment by aniamusial** [comment]
Hi @undercoverGod this is a breaking change, just like @AdamPawlinski said. We try to avoid bc in this version, moving this to 0.12.0. Thanks for reporting! :)
